dynamic_rest/routers.py

```diff
@@ -128,6 +129,87 @@ def register(self, prefix, viewset, base_name=None):
         current[endpoint]['_url'] = url_name
         current[endpoint]['_viewset'] = viewset
 
+    def register_resource(self, viewset, namespace=None):
+        """
+        Register a viewset that should be considered the canonical
+        endpoint for a particular resource. In addition to generating
+        and registering the route, it adds the route in a reverse map
+        to allow DREST to build the canonical URL for a given resource.
+
+        Arguments:
+            viewset - viewset class, should have `serializer_class` attr.
+            namespace - (optional) URL namespace, e.g. 'v3'.
+        """
+
+        # Try to extract resource name from viewset.
+        try:
+            resource_name = viewset.serializer_class().get_plural_name()
+        except:
+            raise Exception(
+                "Failed to extract resource name from viewset: '%s'."
+                " It, or its serializer, may not be DREST-compatible." % (
+                    viewset
+                )
+            )
+
+        # Construct canonical path and register it.
+        if namespace:
+            namespace = namespace.rstrip('/') + '/'
+        base_path = namespace or ''
+        base_path = r'%s' % base_path + resource_name
+        self.register(base_path, viewset)
+
+        # Make sure resource isn't already registered.
+        if resource_name in resource_map:
+            raise Exception(
```

Comment:
For now, I think the idea would be that only one can be mapped as canonical using `register_resource`, though it's possible to register other endpoints using the plain `register` API (i.e. both `/v2/foo/` and `/v3/foo/` can exist, but only one can be canonical). It may become necessary to support canonical resources within versions/namespaces, but in our case, we had `/v2` and `/v3` co-exist for a while, so I think this approach would've worked.